README.md

```diff
@@ -63,19 +83,96 @@ To enable pgvectorscale:
+        document_id BIGINT FOREIGN KEY(document.id)
+        metadata JSONB,
+        contents TEXT,
+        embedding VECTOR(1538)
+    )
+    ```
+
+1. Populate the table any way you like. Please see the [pgvector instructions](https://github.com/pgvector/pgvector/blob/master/README.md#storing) and [list of clients](https://github.com/pgvector/pgvector/blob/master/README.md#languages) for more information.
+1. Create a StreamingDiskANN index on the embedding column
+    ```postgresql
+    CREATE INDEX document_embedding_idx ON document_embedding
+    USING diskann (embedding);
+    ```
+1. Use the following query to find the 10 closest embeddings using the index
+
+    ```postgresql
+    SELECT *
+    FROM document_embedding
+    ORDER BY embedding <=> $1
+    LIMIT 10
+    ```
+
+    Note: as of now we only support cosine distance (`<=>`) queries. Please file an issue to request additional distance types.
+
+## Tunning
+
+The StreamingDiskANN index comes with **smart defaults** but also the ability to customize it's behavior. There are two types of parameters: index build-time parameters that are specified when an index is created and query-time parameters that can be tuned when querying an index.
+
+We suggest setting the index build-time paramers for major changes to index operations while query-time parameters can be used to tune the accuracy/performancy tradeoff for individual queries.
+
+ We expect most people to tune the query-time parameters (if any) and leave the index build time parameters set to default.
+
+### StreamingDiskANN index build-time parameters
+
+These parameters can be set when an index is created.
+
+| Parameter name   | Description                                                                                                                                                    | Default value |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|
+| `storage_layout` | `memory_optimized` which uses SBQ to compress vector data or `plain` which stores data uncompressed | memory_optimized
+| `num_neighbors`    | Sets the maximum number of neighbors per node. Higher values increase accuracy but make the graph traversal slower.                                           | 50            |
+| `search_list_size` | This is the S parameter used in the greedy search algorithm used during construction. Higher values improve graph quality at the cost of slower index builds. | 100           |
+| `max_alpha`        | Is the alpha parameter in the algorithm. Higher values improve graph quality at the cost of slower index builds.                                              | 1.2           |
+| `num_dimensions` | The number of dimensions to index. By default, all dimensions are indexed. But you can also index less dimensions to make use of Matryoshka embeddings | 0 (all dimensions)
```

Comment:
are there min/max ranges for these values? Would be nice to list if so.

Reply:
There aren't good ones right now. Will add a todo